zcash_client_backend/src/data_api.rs

```diff
@@ -1811,9 +1860,32 @@ pub trait WalletWrite: WalletRead {
     #[cfg(feature = "transparent-inputs")]
     fn reserve_next_n_ephemeral_addresses(
         &mut self,
-        account_id: Self::AccountId,
-        n: usize,
-    ) -> Result<Vec<(TransparentAddress, TransparentAddressMetadata)>, Self::Error>;
+        _account_id: Self::AccountId,
+        _n: usize,
+    ) -> Result<Vec<(TransparentAddress, TransparentAddressMetadata)>, Self::Error> {
+        // Default impl is required for feature-flagged trait methods to prevent
+        // breakage due to inadvertent activation of features by transitive dependencies
+        // of the implementing crate.
```

Comment:
Ah. The existing pattern seemed to be that `WalletRead` methods had a default implementation but `WalletWrite` methods did not. Note that `WalletRead::get_orchard_nullifiers` is feature-flagged "orchard" but does not have a default implementation (just as the first example I found; there might be others).

Reply:
I would like to remove all of the noop default impls, but I think that this should be done as a separate commit once this has merged.